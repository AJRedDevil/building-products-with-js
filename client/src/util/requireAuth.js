// npm packages
import React from 'react';
import {Redirect, Route} from 'react-router-dom';
import PropTypes from 'prop-types';

// our packages
import {isAuthenticated} from './checkAuth';

export const PrivateRoute = ({component: Component, ...rest}) => (
  <Route
    {...rest}
    render={props => (
      isAuthenticated() ? (
        <Component {...props} />
      ) : (
        <Redirect to={{
          pathname: '/login',
          state: {from: props.location.pathname},
          }}
        />
      )
    )
    }
  />
);
PrivateRoute.propTypes = {
  component: PropTypes.func.isRequired,
  location: PropTypes.shape({pathname: PropTypes.string}),
};
PrivateRoute.defaultProps = {
  location: {pathname: ''},
};
