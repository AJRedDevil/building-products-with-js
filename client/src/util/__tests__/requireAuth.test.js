// npm packages
import React from 'react';

// our packages
import {PrivateRoute} from '../requireAuth';

test('# PrivateRoute', () => {
  const Test = () => <div />;
  const location = {pathname: 'jpt'};
  const wrapper = shallow(<PrivateRoute component={Test} location={location} />);
  expect(wrapper).toMatchSnapshot();

  localStorage.setItem('user.token', true);
  const wrapperHome = shallow(<PrivateRoute component={Test} />);
  expect(wrapperHome).toMatchSnapshot();
});
