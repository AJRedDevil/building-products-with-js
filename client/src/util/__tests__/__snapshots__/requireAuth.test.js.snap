// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# PrivateRoute 1`] = `
<Route
  location={
    Object {
      "pathname": "jpt",
    }
  }
  render={[Function]}
/>
`;

exports[`# PrivateRoute 2`] = `
<Route
  location={
    Object {
      "pathname": "",
    }
  }
  render={[Function]}
/>
`;
